examples/files.rs

```diff
@@ -0,0 +1,218 @@
+use std::collections::HashSet;
+use std::ffi::OsString;
+use std::path::{Path, PathBuf};
+use std::time::{Duration, Instant};
+
+use crossterm::event::{Event, KeyCode, MouseEventKind};
+use ratatui::backend::{Backend, CrosstermBackend};
+use ratatui::layout::Rect;
+use ratatui::style::{Color, Modifier, Style};
+use ratatui::text::{Line, Span};
+use ratatui::widgets::{Block, Scrollbar, ScrollbarOrientation};
+use ratatui::{Frame, Terminal};
+use tui_tree_widget::{Node, Tree, TreeData, TreeState};
+
+struct FileTreeData(PathBuf);
+
+impl TreeData for FileTreeData {
+    type Identifier = OsString;
+
+    fn flatten(&self, open: &HashSet<Vec<Self::Identifier>>) -> Vec<Node<Self::Identifier>> {
+        let mut result = Vec::new();
+        flatten_inner(&mut result, open, &self.0, &[]);
+        result
+    }
+
+    fn render(
+        &self,
+        identifier: &[Self::Identifier],
+        area: ratatui::layout::Rect,
+        buffer: &mut ratatui::buffer::Buffer,
+    ) {
+        let mut path = self.0.clone();
+        for i in identifier {
+            path = path.join(i);
+        }
+
+        let mut spans = Vec::new();
+
+        if let Some(filename) = path.file_stem() {
+            spans.push(Span::raw(filename.to_string_lossy()));
+        }
+
+        if let Some(extension) = path.extension() {
+            const STYLE: Style = Style::new().fg(Color::DarkGray);
+            spans.push(Span::styled(".", STYLE));
+            spans.push(Span::styled(extension.to_string_lossy(), STYLE));
+        }
+
+        ratatui::widgets::Widget::render(Line::from(spans), area, buffer);
+    }
+}
+
+fn flatten_inner(
```

Comment:
```suggestion
fn flatten_recursive(
```
names this after what it does, rather than where it's used

This could do with a comment about what it does btw for helping those reading the examples find their way.